tasks.py

```diff
@@ -239,13 +240,14 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None, protocol=None, nod
     run("kubectl delete po -nmetallb-system --all", echo=True)
 
     manifests_dir = os.getcwd() + "/manifests"
+    curr_branch = subprocess.check_output("git rev-parse --abbrev-ref HEAD", shell=True).decode().strip("\n")
     with tempfile.TemporaryDirectory() as tmpdir:
         # Copy namespace manifest.
         shutil.copy(manifests_dir + "/namespace.yaml", tmpdir)
 
         with open(manifests_dir + "/metallb.yaml") as f:
             manifest = f.read()
-        manifest = manifest.replace(":main", ":dev-{}".format(architecture))
+        manifest = re.sub(":{}\.?[0-9]?".format(curr_branch), ":dev-{}".format(architecture), manifest)
```

Comment:
I tested out the regex with some different scenarios and I think it will break if the current manifest has a two digit patch release, like `quay.io/metallb/controller:v0.10.10`.  It will leave the last `0` at the end.